src/score_gen/score.rs

```diff
@@ -395,6 +395,16 @@ fn scalar_to_bits(scalar: &Scalar) -> [u8; 256] {
     res
 }
 
+fn next_pow_2_bitnum(mut scalar: Scalar) -> u64 {
```

Comment:
As mentioned we probably want to change the name of the functions, you basically named it by its implementation instead of what it is (counting bits).

I would have something like:

```rust
let num_bits = bits_count(max_range);
```
or something like that.